cernopendata_client/cli.py

```diff
@@ -100,25 +102,67 @@ def get_file_locations(server, recid, doi, title, protocol, expand):
     if recid is not None:
         validate_recid(recid)
     record_json = get_record_as_json(server, recid, doi, title)
-    file_locations = [file["uri"] for file in record_json["metadata"]["files"]]
-    if expand:
-        # let's unwind file indexes
-        file_locations_expanded = []
-        for file in file_locations:
-            if file.endswith("_file_index.txt"):
-                url_file = file.replace("root://eospublic.cern.ch/", server)
-                req = requests.get(url_file)
-                for url_individual_file in req.text.split("\n"):
-                    if url_individual_file:
-                        file_locations_expanded.append(url_individual_file)
-            elif file.endswith("_file_index.json"):
-                pass
-            else:
-                file_locations_expanded.append(file)
-        file_locations = file_locations_expanded
-
-    if protocol == "http":
-        file_locations = [
-            f.replace("root://eospublic.cern.ch/", server) for f in file_locations
-        ]
+    file_locations = get_files_list(server, record_json, protocol, expand)
     click.echo("\n".join(file_locations))
+
+
+@cernopendata_client.command()
+@click.option("--recid", type=click.INT, help="Record ID")
+@click.option("--doi", help="Digital Object Identifier.")
+@click.option("--title", help="Record title")
+@click.option(
+    "--protocol",
+    default="http",
+    type=click.Choice(["root", "http"]),
+    help="Protocol to be used in links.",
+)
+@click.option(
+    "--expand/--no-expand", default=True, help="Expand file indexes? [default=yes]"
+)
+@click.option(
+    "--server",
+    default="http://opendata.cern.ch",
+    type=click.STRING,
+    help="Which CERN Open Data server to query? [default=http://opendata.cern.ch]",
+)
+def download_files(server, recid, doi, title, protocol, expand):
+    """Download the list of files belonging to a dataset."""
+    validate_server(server)
+    if recid is not None:
+        validate_recid(recid)
+    if protocol == "root":
+        click.secho(
+            "Root protocol is not supported yet.",
+            fg="red",
+            err=True,
+        )
+        sys.exit(1)
+    record_json = get_record_as_json(server, recid, doi, title)
+    file_locations = get_files_list(server, record_json, protocol, expand)
+
+    path = str(recid)
+    try:
+        os.mkdir(path)
```

Comment:
Please check first whether the directory does not already exists; there is no need to create it then.  Users would see a message like `Creation of the directory 5500 failed` and might be confused.